src/UIKit/UIApplication.cs

```diff
@@ -57,33 +60,31 @@ internal static void Initialize ()
 		}
 		
 #if !WATCH
-		public static void Main (string [] args, string principalClassName, string delegateClassName)
+		[Obsolete ("Use the overload with 'Type' instead of 'String' parameters for type safety.")]
+		public static void Main (string []? args, string? principalClassName, string? delegateClassName)
 		{
-			var p = NSString.CreateNative (principalClassName);
-			var d = NSString.CreateNative (delegateClassName);
-			try {
-				Main (args, p, d);
-			} finally {
-				// it just looks nicer to release them
-				NSString.ReleaseNative (d);
-				NSString.ReleaseNative (p);
-			}
+			var p = CFString.CreateNative (principalClassName);
+			var d = CFString.CreateNative (delegateClassName);
+			Initialize ();
+			UIApplicationMain (args == null ? 0 : args.Length, args, p, d);
```

Comment:
Other way to write this:
```suggestion
			UIApplicationMain (args?.Length ?? 0, args, p, d);
```